warehouse/integrations/bigquery/bigquery.go

```diff
@@ -163,23 +146,80 @@ func (bq *BigQuery) DeleteTable(ctx context.Context, tableName string) (err erro
 	return
 }
 
+// CreateTable creates a table in BigQuery with the provided schema
+// It also creates a view for the table to deduplicate the data
+// If custom partitioning is enabled, it creates a table with custom partitioning based on the partition column and type
+// only if the partition column exists in the schema.
+// Otherwise, it creates a table with ingestion-time partitioning
 func (bq *BigQuery) CreateTable(ctx context.Context, tableName string, columnMap model.TableSchema) error {
-	bq.logger.Infof("BQ: Creating table: %s in bigquery dataset: %s in project: %s", tableName, bq.namespace, bq.projectID)
+	log := bq.logger.Withn(
+		logger.NewStringField(logfield.ProjectID, bq.projectID),
+		obskit.Namespace(bq.namespace),
+		logger.NewStringField(logfield.TableName, tableName),
+	)
+	log.Infon("Creating table")
+
 	sampleSchema := getTableSchema(columnMap)
-	metaData := &bigquery.TableMetadata{
-		Schema:           sampleSchema,
-		TimePartitioning: &bigquery.TimePartitioning{},
+
+	partitionColumn := bq.partitionColumn()
+	partitionType := bq.partitionType()
+
+	var timePartitioning *bigquery.TimePartitioning
+	if partitionColumn == "" || partitionType == "" {
+		timePartitioning = &bigquery.TimePartitioning{
+			Type: bigquery.DayPartitioningType,
+		}
+	} else {
+		if err := bq.checkValidPartitionColumn(partitionColumn); err != nil {
+			return fmt.Errorf("check valid partition column: %w", err)
+		}
+		bqPartitionType, err := bq.bigqueryPartitionType(partitionType)
+		if err != nil {
+			return fmt.Errorf("bigquery partition type: %w", err)
+		}
+
+		// If partition column is _PARTITIONTIME and partition type is not empty, then we only set the partition type
+		if partitionColumn == "_PARTITIONTIME" {
+			timePartitioning = &bigquery.TimePartitioning{
+				Type: bqPartitionType,
+			}
+		} else {
+			// Checking if the partition column exists in the schema, because in case of
+			// 1. rudder_discards: we only have timestamp column.
+			// 2. rudder_identity_merge_rules: we don't have any column.
+			// 3. rudder_identity_mappings: we don't have any column.
+			_, ok := columnMap[partitionColumn]
+			if ok {
+				log.Infon("Creating table: Partition column found in schema",
+					logger.NewStringField("partitionColumn", partitionColumn),
+					logger.NewStringField("partitionType", partitionType),
+				)
+				timePartitioning = &bigquery.TimePartitioning{
+					Field: partitionColumn,
+					Type:  bqPartitionType,
+				}
+			} else {
+				log.Warnn("Creating table: Partition column not found in schema",
+					logger.NewStringField("partitionColumn", partitionColumn),
+					logger.NewStringField("partitionType", partitionType),
+				)
+				timePartitioning = &bigquery.TimePartitioning{
+					Type: bigquery.DayPartitioningType,
```

Comment:
if the column is not found we fall-back to using `_PARTITIONTIME`

Reply:
Yeah, this is just a safety check.